stake-pool/py/stake_pool/instructions.py

```diff
@@ -917,3 +983,77 @@ def decrease_validator_stake(params: DecreaseValidatorStakeParams) -> Transactio
             )
         )
     )
+
+
+def create_token_metadata(params: CreateTokenMetadataParams) -> TransactionInstruction:
+    """Creates an instruction to create metadata using the mpl token metadata program for the pool token."""
+    if params.withdraw_authority:
+        withdraw_authority = params.withdraw_authority
+    else:
+        (withdraw_authority, _seed) = find_withdraw_authority_program_address(params.program_id, params.stake_pool)
+
+    if params.token_metadata:
+        token_metadata = params.token_metadata
+    else:
+        (token_metadata, _seed) = find_metadata_account(params.pool_mint)
```

Comment:
To avoid the earlier footgun, let's keep this logic as dumb as possible and put *all* parameters in `CreateTokenMetadataParams`, including the rent sysvar, system program id, and metadata program id, and *not* make any of them optional.  

Essentially, let's make this the same as all of the other instruction creators, and then the `action` can fill in the right parameters